traitsui/editors/api.py

```diff
@@ -7,7 +7,7 @@
 try:
     from .array_editor import ArrayEditor
 except ImportError as e:
-    if 'numpy' not in e.message:
```

Comment:
I would think that the check should try to import numpy and on success import ArrayEditor instead of this *ugly* check on the error message. Is there a reason for not doing that?